extensions/lifecycle/objectProcessor/LifecycleObjectProcessor.js

```diff
@@ -134,8 +136,12 @@ class LifecycleObjectProcessor extends EventEmitter {
         async.parallel([
             done => this._setupConsumers(done),
             done => {
-                this._gcProducer = new GarbageCollectorProducer();
-                this._gcProducer.setupProducer(done);
+                if (this._gcConfig) {
```

Comment:
this `gcConfig` is actually not used : so it should not be a `config`, but really just a boolean flag...

....and better yet, this the `gcProducer` is used only in the transition processor : it should actually be moved into that class, there is (theorically) no point keeping it in the base class...